esi_leap/api/controllers/v1/contract.py

```diff
@@ -52,19 +54,29 @@ def __init__(self, **kwargs):
 class ContractsController(rest.RestController):
 
     @wsme_pecan.wsexpose(Contract, wtypes.text)
-    def get_one(self, contract_uuid):
+    def get_one(self, contract_id):
         request = pecan.request.context
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:contract:get', cdict, cdict)
 
-        c = contract.Contract.get(request, contract_uuid)
+        c_id = contract.Contract.get(contract_id)
+        print(c_id)
 
-        if c.project_id != request.project_id:
-            o = offer.Offer.get(request.project_id, c.offer_uuid)
-            if o.project_id != request.project_id:
-                policy.authorize('esi_leap:contract:get_admin', cdict, cdict)
+        permitted = []
+        for c in c_id:
+            try:
+                ContractsController._contract_authorize_management(c, cdict)
+                permitted.append(c)
+            except oslo_policy.PolicyNotAuthorized:
+                continue
 
-        return Contract(**c.to_dict())
+            if len(permitted) > 1:
+                raise exception.ContractDuplicateName(name=contract_id)
+
+        if len(permitted) == 0:
+            raise exception.ContractNotFound(contract_id=contract_id)
```

Comment:
This is definitely a good change - having ContractNotFound instead of an authorization error prevents a potential hacker from trying to figure out existing names simply by randomly trying to `get` a bunch of them.